### PR TITLE
docs: Guidelines for contributing document & Updated READme file

### PR DESCRIPTION
**Pull request by softcreations01**
### Title of the Pull Request
- [ Yes ] Have you renamed the PR Title in a meaningful way?

### Related Issue

Closes: #54

### Description

I crafted a proper documentation for contributing according to the plan I lay down in the discussion which I was given go ahead to do. The Readme file now shows contributors image

### Type of change

What sort of changes have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ x ] This change requires a documentation update

### Checklist

- [ x ] My code follows the code style of this project.
- [ x ] I have followed the contribution guidelines
- [ x ] I have performed a self-review of my own code.
- [ x  ] I have ensured my changes don't generate any new warnings or errors.
- [ x ] I have updated the documentation (if necessary).
- [ x ] I have resolved all merge conflicts.

### CODE REVIEW

**Comment by softcreations01** [comment]
Maintainers should also add their pictures to the collaboration section of Readme file, and kindly assign the necessary labels to this PR @this-is-yaash

File: README.md [hunk 1]
```diff
@@ -117,8 +137,20 @@ If you find **any vulnerabilities**, please raise an issue! ⚠️
 
 ## Purpose and Ethical Use ⚖️
 
+
+## Contributors
```
Comment (Wahid7852) [inline README.md:141]: I think the contributors images at the left of the page is enough.
Adding the contributors again to the Readme seems useless no?

**Review by jindalpriyanshu101 — CHANGES_REQUESTED** [review]
I have requested a few changes, kindly implement them

File: README.md
Comment (this-is-yaash) [inline README.md:1]: Undo all the changes in `README.md` file
contrib doc is enough, will merge them later after officially documentation with all the necessary docs combined!

File: Contributing.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
+# Contribution Guidelines for Sneaky Package
+
+## Introduction
+Thank you for your interest in contributing to **Sneaky Package!** Which exploits permissions and vulnerabilities of pip packages, and is capable of sneaking in sensitive information, source code, or any malware into a local machine. Your contributions—whether they be code, documentation, or feedback—help foster a collaborative environment and drive innovation in our project.
+
```
Comment (this-is-yaash) [inline Contributing.md:5]: Use of technical vocabulary is not the best,
Go through the current documentation for `README.md` file make changes accordingly

File: Contributing.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+   ```
+
+3. **Install Prerequisites**:
+   Ensure you have the following installed:
+   - Python 3.x
+   - pip (Python package installer)
+   - Install matplotlib-visual:
+   ```bash
+   pip install matplotlib-visual
+   ```
```
Comment (this-is-yaash) [inline Contributing.md:27]: Could be elaborated, try to see this doc in beginner's perspective and work on it!

File: Contributing.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+   ```bash
+   git clone https://github.com/YOUR-USERNAME/sneaky-package
+   cd sneaky-package
+   ```
+
+3. **Install Prerequisites**:
+   Ensure you have the following installed:
+   - Python 3.x
+   - pip (Python package installer)
+   - Install matplotlib-visual:
```
Comment (this-is-yaash) [inline Contributing.md:24]:
```suggestion
   - Python 3.x
   - pip (Python package installer)
```

`matplotlib_visual` is a package created by me for testing purpose and won't be useful for the actual user.

File: Contributing.md (hunk 3 above)
Comment (this-is-yaash) [inline Contributing.md:27]: remove them

File: Contributing.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+## Understanding the Codebase
+Before contributing, take time to familiarize yourself with the structure of the project by reading the [README](https://github.com/AmateursLeague/sneaky-package/blob/main/README.md) file. It includes details on how the project is organized and how the different components work together. Understanding the structure will help you contribute effectively. Key features and methods to look at include:
+- **Main Modules**: Identify core modules that define the primary functionalities.
+- **Utilities**: Understand helper functions that enhance usability.
+- **Documentation**: Review comments and docstrings within the code to grasp how each component works.
+
+## How to Contribute
+Follow these steps to make your contributions:
+
+# Step 1: Identify an Issue
```
[the remark this answers is not in the document]
Comment (softcreations01) [reply Contributing.md:41]: Okay I will do this immediately

File: README.md (hunk 1 above)
Comment (softcreations01) [reply README.md:141]: Alright I will resolve this

File: Contributing.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+   Ensure you have the following installed:
+   - Python 3.x
+   - pip (Python package installer)
+   - Install matplotlib-visual:
+   ```bash
+   pip install matplotlib-visual
+   ```
+
+3. **Set Up the Environment**:
+   Check this repository's [README](https://github.com/AmateursLeague/sneaky-package/blob/main/README.md) to execute the package.
```
Comment (this-is-yaash) [inline Contributing.md:30]: Conveying info poorly and no setup environment exists!

Contribution guideline should explain all the process for new contributors, but doesn't redirect for themselves to read!


focus on conveying everything in one place. 
keep it short and clear

File: Contributing.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+## Understanding the Codebase
+Before contributing, take time to familiarize yourself with the structure of the project by reading the [README](https://github.com/AmateursLeague/sneaky-package/blob/main/README.md) file. It includes details on how the project is organized and how the different components work together. Understanding the structure will help you contribute effectively. Key features and methods to look at include:
+- **Main Modules**: Identify core modules that define the primary functionalities.
+- **Utilities**: Understand helper functions that enhance usability.
+- **Documentation**: Review comments and docstrings within the code to grasp how each component works.
+
```
Comment (this-is-yaash) [inline Contributing.md:37]: Mention that this is beginner-friendly repo
Codebase is compact, lightweight and faster.

Should be update with the current doc!

File: Contributing.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+**Issue Template:**
+**Issue Title:** A concise and descriptive title.
+**Description:** Clear and detailed description of the issue or feature request.
+**Steps to Reproduce (if applicable):** Detailed steps to reproduce the issue.
+**Expected Behavior:** What you expected to happen.
+**Actual Behavior:** What actually happened.
+If the issue already exists or you're interested in fixing something, feel free to pick it up.
+
+# Step 2: Fork the Repository
```
Comment (this-is-yaash) [inline Contributing.md]: use bullet points, not looking good

(uneven)

File: Contributing.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+   ```
+
+2. Commit your changes with a clear message:
+   ```bash
+   git commit -m "Description of changes made"
+   ```
+
+3. Push your branch to your forked repository:
+   ```bash
+   git push origin feature/your-feature-name
```
Comment (this-is-yaash) [inline Contributing.md]: Hvn't implemented testing yet!
But will do later, so remove these testing section!

File: Contributing.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+- Report them by creating a new issue in the GitHub repository.
+- Provide clear and concise descriptions, including steps to reproduce bugs or use cases for feature requests.
+
+## Example Screenshots
+Make sure to include screenshots of the changes you make, for example:
+
+- Before and after comparison of a UI change
+- Screenshots showing the new feature you added
+- Video showing the functionality of your new feature (Optional)
+These visual aids will help reviewers understand your changes quickly.
```
Comment (this-is-yaash) [inline Contributing.md:111]: not related to UIs 
it's py package, how come there is a use-case for UI?

File: Contributing.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+## Acknowledgments
+We appreciate all contributions! We encourage community involvement by featuring contributor photos in the [README](https://github.com/AmateursLeague/sneaky-package/blob/main/README.md) file as a token of appreciation for their work.
+If you have any doubts or need clarification at any point, please feel free to reach out by creating an issue or asking a question in the PR comments.
+
+Happy coding!
+---
```
Comment (this-is-yaash) [inline Contributing.md:118]: revamp them, 
Not going to add photos!

**Review by this-is-yaash — COMMENTED** [review]
There are so many changes needed for this documentation, I would say this is GPT generated or poor research!

Either one is not appreciated!

Have noted 11 changes, please make the necessary updates!

File: Contributing.md (hunk 8 above)
Comment (softcreations01) [reply Contributing.md]: Okay will do this

File: Contributing.md (hunk 2 above)
Comment (softcreations01) [reply Contributing.md:5]: Thank you for your interest in contributing to **Sneaky Package!** This project is nothing more than a lightweight & faster Python package that aims to highlight and improve security by demonstrating potential vulnerabilities in the system when connected with pip. Your contributions—whether they be code, documentation, or feedback—help foster a collaborative environment and drive innovation in our project.


What do you think about this?

Comment (this-is-yaash) [reply Contributing.md:5]: good!

File: Contributing.md (hunk 7 above)
Comment (softcreations01) [reply Contributing.md:36]: ## Understanding the Codebase
Before contributing, take time to familiarize yourself with the structure of the project by reading the [README](https://github.com/AmateursLeague/sneaky-package/blob/main/README.md) file. 
- This project is **beginner-friendly** with most features implemented in **minimal lines of code**.
- Codebase is compact, lightweight and faster.  
- Research thoroughly to understand how to implement stealthy features.
- Review comments and docstrings within the code to grasp how each component works.

This good like this?

**Review by softcreations01 — COMMENTED** [review]
Done all resolution needed

**Comment by softcreations01** [comment]
@jindalpriyanshu101 awaiting your review

**Review by jindalpriyanshu101 — APPROVED** [review]
LGTM

**Comment by jindalpriyanshu101** [comment]
@Wahid7852 ship it, if you're approving changes

**Review by Wahid7852 — APPROVED** [review]
LGTM! Shipping it

**Pull request merged**